url21b471891f_Agetac_Model_tgz-pJ8-org_json_TestJ8/src/org/agetac/common/model/impl/Position.java

```diff
@@ -0,0 +1,72 @@
+package org.agetac.common.model.impl;
+
+import org.agetac.common.exception.InvalidJSONException;
+import org.agetac.common.model.sign.IJsonable;
+import org.json.JSONException;
+import org.json.JSONObject;
+
+public class Position implements IJsonable {
+	
+	private double longitude;
+	private double latitude;
+	
+	public Position() {
+		longitude = 0;
+		latitude = 0;
+	}
+
+	public Position(double longitude, double latitude) {
+		this.longitude = longitude;
+		this.latitude = latitude;
+	}
+	public Position(Position p) {
+		this.longitude = p.getLongitude();
+		this.latitude = p.getLatitude();
+	}
+	
+	public Position(JSONObject json) throws InvalidJSONException{
+		try {
+			this.longitude = json.getDouble("longitude");
+			this.latitude = json.getDouble("latitude");
+		} catch (JSONException e) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: While wrapping the caught exception into a custom one, information about the caught exception is being lost, including information about the stack trace of the exception.

**Fix**: If the caught exception object does not contain sensitive information, consider passing it as the &#34;rootCause&#34; or inner exception parameter to the constructor of the new exception before throwing the new exception. (Note that not all exception constructors support inner exceptions. Use a wrapper exception that supports inner exceptions.) 
[Learn more](https://www.ibm.com/support/pages/best-practice-catching-and-re-throwing-java-exceptions)